source/server/http/admin.cc

```diff
@@ -697,6 +697,18 @@ Http::Code AdminImpl::handlerMemory(absl::string_view, Http::HeaderMap& response
   return Http::Code::OK;
 }
 
+Http::Code AdminImpl::handlerDrainListeners(absl::string_view url, Http::HeaderMap&,
+                                            Buffer::Instance& response, AdminStream&) {
+  const Http::Utility::QueryParams params = Http::Utility::parseQueryString(url);
+  const bool inbound_only = params.find("inboundonly") != params.end();
+  if (server_.listenerManager().shutdownListeners(inbound_only)) {
+    response.add("OK\n");
+  } else {
+    response.add("No listeners matched\n");
```

Comment:
Nit: I'm not sure if it really matters whether we matched any listeners or not. I'm fine leaving it as-is, but pointing this out in case anyone else cares.

Reply:
+1 I'm not sure if it's worth it. Can we remove this and then make the function return void again?